components/psst/browser/core/psst_rule.cc

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/psst/browser/core/psst_rule.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/files/file_path.h"
+#include "base/json/json_reader.h"
+#include "base/types/expected.h"
+#include "extensions/common/url_pattern.h"
```

Comment:
I thought we were not going to use extensions url_pattern here. We want to use content settings patterns instead per the discussion we had with @pes10k

Reply:
This definitely seems like an enhancement.

Reply:
Something we decided to do for this feature doesn't seem like an enhancement to me? In any case I'm fine with moving this to a follow-up, but I think it's important that we switch over to content settings patterns and also the changes suggested by @cdesouza-chromium and making some kind of template so we don't keep duplicating this same code over and over again

Reply:
To clarify, I did try to use URLPattern that you shared in chat but ran into problems, so reverted to this. I think it would be good to change all the components to use CSP in one swoop, I just think this PR is already doing non-feature-specific work (which is good!) but it's already ballooned way beyond the "quick easy MVP PR" it was supposed to be, so I really want to merge it in ASAP once things look safe, especially if it's going to be used by other priority features as well.

Reply:
Can you open an issue for this? It's fine for a follow-up as I mentioned above

Reply:
https://github.com/brave/brave-browser/issues/34060